vars/DocMultiplicationPipeline.groovy

```diff
@@ -0,0 +1,413 @@
+                                        echo "  cd test/tmp/amazon-opensearch-service-sample-cdk"
+                                        echo "  cdk destroy '*' --force"
+                                        echo ""
+                                        echo "Or use the cleanup script:"
+                                        echo "  ./awsSourceClusterSetup.sh --cleanup --cluster-version ${params.engineVersion} --stage ${params.stage} --region ${params.region}"
+                                    """
+                                }
+                                
+                                // Third: Clean up any orphaned bootstrap stacks if they exist
+                                echo "Checking for orphaned bootstrap stacks..."
+                                sh """
+                                    echo "Checking for MigrationBootstrap stack..."
+                                    if aws cloudformation describe-stacks --stack-name "MigrationBootstrap" --region ${params.region} >/dev/null 2>&1; then
+                                        echo "Found MigrationBootstrap stack, cleaning up..."
+                                        aws cloudformation delete-stack --stack-name "MigrationBootstrap" --region ${params.region}
+                                        echo "MigrationBootstrap stack deletion initiated"
+                                    else
+                                        echo "No MigrationBootstrap stack found"
+                                    fi
+                                """
+                            }
+                        }
+                        
+                        echo "CDK-based cleanup completed successfully"
+                        echo "Migration Assistant infrastructure destroyed using proper CDK commands"
+                        echo "Source cluster preserved for reuse (saves 20+ minutes on next run)"
+                    }
+                }
+            }
+            
+            // Success handling
+            echo "SUCCESS: Document Multiplication Test completed successfully!"
+            echo ""
+            echo "Test Results Summary:"
+            echo "  Source Cluster: ${params.clusterVersion} (${params.engineVersion})"
+            echo "  Index Created: ${params.indexName} with ${params.numShards} shards"
+            echo "  Documents Multiplied: ${params.multiplicationFactor}x factor"
+            echo "  Large Snapshot Created: Available in S3"
+            echo ""
+            echo "S3 Location: ${params.snapshotBucketPrefix}${params.stage}/${params.s3DirectoryPrefix}*"
+            echo ""
+            if (!params.skipCleanup) {
+                echo "Resources cleaned up automatically"
+            } else {
+                echo "Cleanup skipped - manual cleanup required"
+            }
+            
+        } catch (Exception e) {
```

Comment:
This is odd. Why aren't we using a cleanup step that always gets ran for any cleanups?